### PR TITLE
ENG-3605 amend entando hub rest api with temp command

### PR DESCRIPTION
**Pull request by jeffgo10**
~- added temporary command `entando-bundle-cli hub-group-api` only for testing purposes.~
~- nocks not yet added due to unknown issue on error "Mocks not yet satisfied"~

~try these examples:~
~- `entando-bundle-cli hub-group-api list`~
~- `entando-bundle-cli hub-group-api bundle --name="Test germano 1"`~
~- `entando-bundle-cli hub-group-api bundle --versionId=51`~

in order to test, please run `npm test`

### CODE REVIEW

File: src/api/hub.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+
+export default class HubGroupAPI {
+  private readonly defaultBaseUrl = 'https://63d6be0e-bd49-4c76-8fe5-195bb7cf88a5.mock.pstmn.io';
```
Comment (cecchisandrone) [inline src/api/hub.ts]: This is not useful here, please remove it

File: src/api/hub.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  private readonly baseUrl: string;
+
+  private readonly apiPath = '/ent/api/templates/bundlegroups';
+
+  constructor(baseUrl?: string) {
```
Comment (cecchisandrone) [inline src/api/hub.ts]: baseUrl required

File: src/api/hub.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  async callApi(uri: string): Promise<any[]> {
+    const response = await axios(`${this.baseUrl}${uri}`);
+    const { data } = response;
+
+    return data;
+  }
+
+  getHubGroups(): Promise<any[]> {
```
Comment (cecchisandrone) [inline src/api/hub.ts]: 1) getBundleGroups
2) This should return BundleGroup[]

File: src/api/hub.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    return this.callApi(this.apiPath);
+  }
+
+  getHubGroupById(versionId: number): Promise<any[]> {
```
Comment (cecchisandrone) [inline src/api/hub.ts]: 1) getBundleGroupById
2) This should return Bundle[]

File: src/api/hub.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    return this.callApi(`${this.apiPath}/${versionId}`);
+  }
+
+  getHubGroupByName(name: string): Promise<any[]> {
```
Comment (cecchisandrone) [inline src/api/hub.ts]: this should be merged with getBundleGroups, we don't need two methods, just an optional name parameter to pass to api. Also use this form:
```
// Equivalent to `axios.get('https://httpbin.org/get?answer=42')`
const res = await axios.get('https://httpbin.org/get', { params: { answer: 42 } });
```

File: src/commands/hub-group-api.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import { CliUx, Command, Flags } from "@oclif/core"
+import HubGroup from '../api/hub'
+
+export default class HubGroupApi extends Command {
```
Comment (cecchisandrone) [inline src/commands/hub-group-api.ts]: we can remove this class, unit test should be enough

File: test/commands/hub-group-api.test.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import { expect, test } from '@oclif/test'
```
Comment (cecchisandrone) [inline test/commands/hub-group-api.test.ts]: move this test to the proper folder (test/api)

File: test/commands/hub-group-api.test.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+const demoBundleOut = {
+  bundleGroupName: "Test germano 1",
+  bundleName: "{{$randomUserName}}",
+  gitSrcRepoAddress: "https://github.com/germano/mysrcrepo2.git",
+  bundleGroupVersionId: 51,
+  bundleGroupId: 49,
+  bundleId: 51,
+};
+
+const demoBundleList = [
```
Comment (cecchisandrone) [inline test/commands/hub-group-api.test.ts]: This is `demoBundleGroupList`

File: test/commands/hub-group-api.test.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import { expect, test } from '@oclif/test'
+
+const demoBundleOut = {
```
Comment (cecchisandrone) [inline test/commands/hub-group-api.test.ts]: This should be `demoBundle`

File: src/api/hub.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+
+export default class HubGroupAPI {
```
Comment (cecchisandrone) [inline src/api/hub.ts]: Just `HubApi`

File: src/api/hub.ts (hunk 1 above)
Comment (jeffgo10) [reply src/api/hub.ts]: fyi, as per our discussion last week, default base url will be retained. So I'll resolve this conversation

File: src/api/hub.ts (hunk 3 above)
Comment (jeffgo10) [reply src/api/hub.ts]: type `BundleGroup` has been created, I have created `BundleGroupId` as well for items with only `BundleGroupName` and `BundleGroupVersionId` properties (in this method for instance)

File: src/api/hub.ts (hunk 4 above)
Comment (jeffgo10) [reply src/api/hub.ts]: is it `Bundle` or `BundleGroup` for the type name?

File: src/api/hub.ts (hunk 5 above)
Comment (jeffgo10) [reply src/api/hub.ts]: done; merged to `getBundleGroups`

File: src/commands/hub-group-api.ts (hunk 6 above)
Comment (jeffgo10) [reply src/commands/hub-group-api.ts]: removed

File: test/commands/hub-group-api.test.ts (hunk 7 above)
Comment (jeffgo10) [reply test/commands/hub-group-api.test.ts]: done; file is also renamed to `hub-api.test.ts`

File: test/commands/hub-group-api.test.ts (hunk 8 above)
Comment (jeffgo10) [reply test/commands/hub-group-api.test.ts]: done

File: test/commands/hub-group-api.test.ts (hunk 9 above)
Comment (jeffgo10) [reply test/commands/hub-group-api.test.ts]: done

File: src/api/hub.ts (hunk 10 above)
Comment (jeffgo10) [reply src/api/hub.ts]: done; file renamed to `hub-api`

File: src/api/hub.ts (hunk 4 above)
Comment (cecchisandrone) [reply src/api/hub.ts]: the return should be `Bundle[]`

File: src/api/hub-api.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  async callApi(uri: string, params?: BundleGroupAPIParam): Promise<any[]> {
+    const response = await axios(`${this.baseUrl}${uri}`, params ? { params } : {})
+    const { data } = response
+
+    return data
+  }
+
+  getBundleGroups(name?: string): Promise<BundleGroupId[]> {
```
Comment (cecchisandrone) [inline src/api/hub-api.ts]: this should be `BundleGroup[]`

File: src/api/hub-api.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    return this.callApi(this.apiPath, name ? { name } : undefined)
+  }
+
+  getBundleGroupById(versionId: number): Promise<BundleGroup[]> {
```
Comment (cecchisandrone) [inline src/api/hub-api.ts]: this should be `getBundlesByBundleGroupId(versionId: number): Promise<Bundle[]>`

**Review by cecchisandrone — CHANGES_REQUESTED** [review]

File: src/api/hub.ts (hunk 4 above)
Comment (jeffgo10) [reply src/api/hub.ts]: done

File: src/api/hub-api.ts (hunk 11 above)
Comment (jeffgo10) [reply src/api/hub-api.ts]: done

File: src/api/hub-api.ts (hunk 12 above)
Comment (jeffgo10) [reply src/api/hub-api.ts]: done

**Review by cecchisandrone — APPROVED** [review]

**Pull request merged**